packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -112,8 +112,18 @@ export abstract class DeployExecutor<T> extends DeployRetrieveExecutor<T> {
     components: ComponentSet,
     token: vscode.CancellationToken
   ): Promise<DeployResult | undefined> {
+    const projectPath = getRootWorkspacePath();
+    const connection = await WorkspaceContext.getInstance().getConnection();
+
+    components.projectDirectory = projectPath;
```

Comment:
This change was made to potentially fix the issue where a conflict wasn't being detected during Jeff's initial QA. 

It is also a potential fix for #4775.

And it is a property that should be set on the ComponentSet before deploying/retrieving anyway - see https://github.com/forcedotcom/source-deploy-retrieve/pull/929